# Exploratory Data Analysis on the Movies Dataset

**Importing relevant libraries**

In [91]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


**loading the dataset from the year 2000**

In [93]:
df_2000 = pd.read_csv("C:/Users/heill/OneDrive/Documents/Github/Database_Movies/Movies_files/final_tmdb_data_2000.csv.gz")
df_2000.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres  \
0         NaN                                                NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   

                               homepage        id original_language  \
0                                   NaN       NaN               NaN   
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   

    original_title  ...     revenue  runtime  \
0              NaN  ...         NaN      NaN   
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0           5.4       21.0           NaN  
2    0.0           4.4        7.0           NaN  
3    0.0           0.0        0.0           NaN  
4    0.0           8.1     1860.0            PG  

[5 rows x 26 columns]

**Inspection of dataset**

In [94]:
df_2000.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1192 entries, 0 to 1191
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                1192 non-null   object 
 1   adult                  1191 non-null   float64
 2   backdrop_path          605 non-null    object 
 3   belongs_to_collection  107 non-null    object 
 4   budget                 1191 non-null   float64
 5   genres                 1191 non-null   object 
 6   homepage               65 non-null     object 
 7   id                     1191 non-null   float64
 8   original_language      1191 non-null   object 
 9   original_title         1191 non-null   object 
 10  overview               1167 non-null   object 
 11  popularity             1191 non-null   float64
 12  poster_path            1047 non-null   object 
 13  production_companies   1191 non-null   object 
 14  production_countries   1191 non-null   object 
 15  rele

**Loading dataset from for the year 2001**

In [95]:
df_2001 = pd.read_csv("C:/Users/heill/OneDrive/Documents/Github/Database_Movies/Movies_files/final_tmdb_data_2001.csv.gz")
df_2001.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423    0.0  /hfeiSfWYujh6MKhtGTXyK3DD4nN.jpg                   NaN   
2  tt0114447    0.0                               NaN                   NaN   
3  tt0118589    0.0  /bLntSfsqUheiUirdV9Ts8znqsTM.jpg                   NaN   
4  tt0118652    0.0  /mWxJEFRMvkG4UItYJkRDMgWQ08Y.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  48000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 14, 'nam...      NaN   
2         0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN   
3  22000000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   
4   1000000.0  [{'id': 9648, 'name': 'Mystery'}, {'id': 27, '...      NaN   

         id original_language         original_title  ...     revenue  \
0       NaN               NaN                    NaN  ...         NaN   
1   11232.0                en         Kate & Leopold  ...  76019048.0   
2  151007.0                en       The Silent Force  ...         0.0   
3   10696.0                en                Glitter  ...   5271666.0   
4   17140.0                en  The Attic Expeditions  ...         0.0   

   runtime                                   spoken_languages    status  \
0      NaN                                                NaN       NaN   
1    118.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2     90.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3    104.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4    100.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline                  title  \
0                                                NaN                    NaN   
1  If they lived in the same century they'd be pe...         Kate & Leopold   
2  They left him for dead... They should have fin...       The Silent Force   
3   In music she found her dream, her love, herself.                Glitter   
4  His search for peace of mind... will leave his...  The Attic Expeditions   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0           6.3     1103.0         PG-13  
2    0.0           5.0        3.0           NaN  
3    0.0           4.6      115.0         PG-13  
4    0.0           5.0       27.0             R  

[5 rows x 26 columns]

**Inspection of dataset**

In [96]:
df_2001.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 920 entries, 0 to 919
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                920 non-null    object 
 1   adult                  919 non-null    float64
 2   backdrop_path          484 non-null    object 
 3   belongs_to_collection  57 non-null     object 
 4   budget                 919 non-null    float64
 5   genres                 919 non-null    object 
 6   homepage               81 non-null     object 
 7   id                     919 non-null    float64
 8   original_language      919 non-null    object 
 9   original_title         919 non-null    object 
 10  overview               896 non-null    object 
 11  popularity             919 non-null    float64
 12  poster_path            820 non-null    object 
 13  production_companies   919 non-null    object 
 14  production_countries   919 non-null    object 
 15  releas

**Joining the two dataset into one dataframe**

In [169]:
df =pd.concat([df_2000, df_2001], axis = 0)
df.head(3)

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   

         id original_language   original_title  ... revenue  runtime  \
0       NaN               NaN              NaN  ...     NaN      NaN   
1   62127.0                en  The Fantasticks  ...     0.0     86.0   
2  110977.0                en    For the Cause  ...     0.0    100.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                         tagline            title  video  \
0                                            NaN              NaN    NaN   
1  Try to remember the first time magic happened  The Fantasticks    0.0   
2   The ultimate showdown on a forbidden planet.    For the Cause    0.0   

   vote_average vote_count certification  
0           NaN        NaN           NaN  
1           5.4       21.0           NaN  
2           4.4        7.0           NaN  

[3 rows x 26 columns]

**Inspection of the joint dataframe**

In [98]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2112 entries, 0 to 919
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                2112 non-null   object 
 1   adult                  2110 non-null   float64
 2   backdrop_path          1089 non-null   object 
 3   belongs_to_collection  164 non-null    object 
 4   budget                 2110 non-null   float64
 5   genres                 2110 non-null   object 
 6   homepage               146 non-null    object 
 7   id                     2110 non-null   float64
 8   original_language      2110 non-null   object 
 9   original_title         2110 non-null   object 
 10  overview               2063 non-null   object 
 11  popularity             2110 non-null   float64
 12  poster_path            1867 non-null   object 
 13  production_companies   2110 non-null   object 
 14  production_countries   2110 non-null   object 
 15  relea

In [195]:
df.tail(500)


imdb_id  adult                     backdrop_path belongs_to_collection  \
420  tt0249624    0.0  /1pEN1mplx9lg1Y9XeFkGymiDKKD.jpg                   NaN   
421  tt0250081    0.0  /oYUOT1Ijrpy5cDcyNdlQw1sTfvY.jpg                   NaN   
422  tt0250125    0.0                               NaN                   NaN   
423  tt0250202    0.0  /dJxCfBDW1VuwNi0ZyiTgJcOgvNH.jpg                   NaN   
424  tt0250224    0.0  /zbFG63ineDZdEHHyv7WwASRpOAM.jpg                   NaN   
..         ...    ...                               ...                   ...   
915  tt0288602    0.0  /kaibkGad02ufy8pgIZrT7yUmIOz.jpg                   NaN   
916  tt0288626    0.0                               NaN                   NaN   
917  tt0288719    0.0                               NaN                   NaN   
918  tt0288869    0.0                               NaN                   NaN   
919  tt0289054    0.0  /2f8IywATs84lgKsKVfbtO6eKiRJ.jpg                   NaN   

         budget                                             genres homepage  \
420         0.0                      [{'id': 18, 'name': 'Drama'}]      NaN   
421         0.0  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...      NaN   
422         0.0                      [{'id': 18, 'name': 'Drama'}]      NaN   
423         0.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...      NaN   
424  13000000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   
..          ...                                                ...      ...   
915         0.0                                                 []      NaN   
916         0.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...      NaN   
917         0.0                      [{'id': 18, 'name': 'Drama'}]      NaN   
918         0.0                                                 []      NaN   
919         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

           id original_language               original_title  ...     revenue  \
420  286369.0                en                 Gabriel & Me  ...         0.0   
421   16550.0                en                 Storytelling  ...         0.0   
422   44081.0                fa                 زیر پوست شهر  ...         0.0   
423   21055.0                en             All Over the Guy  ...   1051948.0   
424   10691.0                en              Crazy/Beautiful  ...  16929123.0   
..        ...               ...                          ...  ...         ...   
915  692178.0                es  Arregui, la noticia del día  ...         0.0   
916  447111.0                el            Μια μέρα τη νύχτα  ...         0.0   
917  291592.0                es           Los pasos perdidos  ...         0.0   
918  126975.0                en               Vidas privadas  ...         0.0   
919   44768.0                ja                   赤い橋の下のぬるい水  ...         0.0   

     runtime                                   spoken_languages    status  \
420     84.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
421     87.0  [{'english_name': 'German', 'iso_639_1': 'de',...  Released   
422     92.0  [{'english_name': 'Persian', 'iso_639_1': 'fa'...  Released   
423     95.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
424     99.0  [{'english_name': 'Spanish', 'iso_639_1': 'es'...  Released   
..       ...                                                ...       ...   
915    103.0  [{'english_name': 'Spanish', 'iso_639_1': 'es'...  Released   
916    110.0  [{'english_name': 'Greek', 'iso_639_1': 'el', ...  Released   
917      0.0  [{'english_name': 'Spanish', 'iso_639_1': 'es'...  Released   
918     97.0  [{'english_name': 'Spanish', 'iso_639_1': 'es'...  Released   
919    119.0  [{'english_name': 'Japanese', 'iso_639_1': 'ja...  Released   

                                               tagline  \
420                                                NaN   
421  From the director of Happiness and Welcome To ...   
422          

## How many movies had at least some valid financial information (values > 0 for budget OR revenue)?

In [158]:
#Grouping the dataframe based on budget and revenue with values greater than 0
new_df = df[(df["budget"]>0) & (df["revenue"]>0) ]
new_df.head()

imdb_id  adult                     backdrop_path  \
4   tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
11  tt0120467    0.0                               NaN   
13  tt0120630    0.0  /t7aJ7hqsMLEoqBJk7yAXxcAVvcz.jpg   
16  tt0120753    0.0  /cXXxi0j9sXFt7LbeajuFoKFqWMu.jpg   
17  tt0120755    0.0  /8XaohdSqoBEWHhW40q2VTdMBrV0.jpg   

                                belongs_to_collection       budget  \
4                                                 NaN     150000.0   
11                                                NaN     120000.0   
13  {'id': 718551, 'name': 'Chicken Run Collection...   45000000.0   
16                                                NaN    8000000.0   
17  {'id': 87359, 'name': 'Mission: Impossible Col...  125000000.0   

                                               genres  \
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
11  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...   
13  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
16  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...   
17  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   

                                homepage       id original_language  \
4   http://www.wkw-inthemoodforlove.com/    843.0                cn   
11                                   NaN  19085.0                en   
13                                   NaN   7443.0                en   
16                                   NaN    318.0                en   
17                                   NaN    955.0                en   

              original_title  ...      revenue  runtime  \
4                       花樣年華  ...   12854953.0     99.0   
11                    Vulgar  ...      14904.0     87.0   
13               Chicken Run  ...  224834564.0     84.0   
16  The Million Dollar Hotel  ...     105983.0    122.0   
17    Mission: Impossible II  ...  546388105.0    123.0   

                                     spoken_languages    status  \
4   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
11  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
13  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
16  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
17  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                              tagline  \
4   Feel the heat, keep the feeling burning, let t...   
11   Everyone loves a clown... some more than others.   
13  This ain't no chick flick. It's poultry in mot...   
16                    Everyone has something to hide.   
17                       Expect the impossible again.   

                       title  video  vote_average vote_count certification  
4       In the Mood for Love    0.0           8.1     1860.0            PG  
11                    Vulgar    0.0           5.5       40.0             R  
13               Chicken Run    0.0           6.7     3992.0             G  
16  The Million Dollar Hotel    0.0           5.9      247.0             R  
17    Mission: Impossible II    0.0           6.1     5266.0         PG-13  

[5 rows x 26 columns]

**There are only 338 movies with information on budget or revenue**

In [173]:
new_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 338 entries, 4 to 889
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                338 non-null    object 
 1   adult                  338 non-null    float64
 2   backdrop_path          328 non-null    object 
 3   belongs_to_collection  83 non-null     object 
 4   budget                 338 non-null    float64
 5   genres                 338 non-null    object 
 6   homepage               52 non-null     object 
 7   id                     338 non-null    float64
 8   original_language      338 non-null    object 
 9   original_title         338 non-null    object 
 10  overview               338 non-null    object 
 11  popularity             338 non-null    float64
 12  poster_path            338 non-null    object 
 13  production_companies   338 non-null    object 
 14  production_countries   338 non-null    object 
 15  releas

## How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [125]:
#Number of movies per certification
new_df["certification"].value_counts()

R        138
PG-13    116
PG        29
G         13
NR         6
Name: certification, dtype: int64

In [126]:
#Total number of movies with certification under the new filtered dataframe
new_df["certification"].value_counts().sum()

302

In [128]:
#Number of movies with null values in certification column

new_df["certification"].isnull().sum()

36

**R certification has 138 movies, PG-13 has 116, PG has 29 movies, G has a total of 13 and NR has only 16 movies. The dataframe also has 36 "certification" columns with null values , which sums up to the total number of enteries for the filtered dataframe(338).**

## What is the average revenue per certification category?

In [164]:
#Grouping the dataframe based on certification and extracting the average revenue per certification
filt =round(new_df.groupby(['certification']).mean())
filt['revenue']=filt['revenue'].apply('{:,}'.format)
filt["revenue"]


certification
G        133,216,882.0
NR        23,823,316.0
PG       133,213,222.0
PG-13    111,495,504.0
R         53,924,800.0
Name: revenue, dtype: object

## What is the average budget per certification category?

In [165]:
#Grouping the dataframe based on certification and extracting the average budget per certification
filt =round(new_df.groupby(['certification']).mean())
filt['budget']=filt['budget'].apply('{:,}'.format)
filt["budget"]

certification
G        44,000,000.0
NR       14,783,333.0
PG       53,741,380.0
PG-13    47,081,012.0
R        27,625,779.0
Name: budget, dtype: object

**Saving the combined dataframe into an existing folder(FOLDER)"

In [92]:
#Checking the contents of the folder
import os
os.listdir(FOLDER)

['final_tmdb_data_2000.csv.gz',
 'final_tmdb_data_2001.csv.gz',
 'New folder',
 'tmdb_api_results_2000.json',
 'tmdb_api_results_2001.json']

In [177]:
#saving the dataframe
df.to_csv(f"{FOLDER}tmdb_results_combined.csv.gz",  index =False)

In [179]:
#Confirmig the success of the saving process
os.listdir(FOLDER)

['final_tmdb_data_2000.csv.gz',
 'final_tmdb_data_2001.csv.gz',
 'New folder',
 'tmdb_api_results_2000.json',
 'tmdb_api_results_2001.json',
 'tmdb_results_combined.csv.gz']